Resend "Split ATen/Parallel into interface and backend"

ilia-cher opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #20157 Support for Eigen thread pool
* #20480 Native TBB parallel backend
* #20454 Restore TBB module
* #20087 Native ATen/Parallel backend
* **#20825 Resend "Split ATen/Parallel into interface and backend"**
* #20848 Fix init_thread calls in thread pool initialization

Summary:
Resending https://github.com/pytorch/pytorch/pull/20057

Test Plan:
https://github.com/pytorch/pytorch/pull/20057

Differential Revision: [D15458073](https://our.internmc.facebook.com/intern/diff/D15458073)

Commit:
Resend "Split ATen/Parallel into interface and backend"

Summary:
Resending https://github.com/pytorch/pytorch/pull/20057

Test Plan:
https://github.com/pytorch/pytorch/pull/20057

ezyang commented:
Unfortunately you will need to rebase this past  fdb9239 because there was more breakage on master

Commit:
Update on "Resend "Split ATen/Parallel into interface and backend""

Resend "Split ATen/Parallel into interface and backend"

Summary:
Resending https://github.com/pytorch/pytorch/pull/20057

Test Plan:
https://github.com/pytorch/pytorch/pull/20057

gh-metadata: pytorch pytorch 20825 gh/ilia-cher/21/head

ilia-cher commented:
@ezyang rebased

ilia-cher commented:
the only two faiures in CI doesn't look related (one failed with 'python: can't open file '.jenkins/pytorch/print_sccache_log.py': [Errno 2] No such file or directory' and another failed with 2 tests failing with 'No module named symbolic_helper' )

Commit:
Update on "Resend "Split ATen/Parallel into interface and backend""

Resend "Split ATen/Parallel into interface and backend"

Summary:
Resending https://github.com/pytorch/pytorch/pull/20057

Test Plan:
https://github.com/pytorch/pytorch/pull/20057

gh-metadata: pytorch pytorch 20825 gh/ilia-cher/21/head

Commit:
Update on "Resend "Split ATen/Parallel into interface and backend""

Resend "Split ATen/Parallel into interface and backend"

Summary:
Resending https://github.com/pytorch/pytorch/pull/20057

Test Plan:
https://github.com/pytorch/pytorch/pull/20057

gh-metadata: pytorch pytorch 20825 gh/ilia-cher/21/head

Commit:
Update on "Resend "Split ATen/Parallel into interface and backend""

Resend "Split ATen/Parallel into interface and backend"

Summary:
Resending https://github.com/pytorch/pytorch/pull/20057

Test Plan:
https://github.com/pytorch/pytorch/pull/20057

gh-metadata: pytorch pytorch 20825 gh/ilia-cher/21/head
